examples/keras_recipes/creating_tfrecords.py

```diff
@@ -0,0 +1,290 @@
+"""
+Title: Creating TFRecords
+Author: [Dimitre Oliveira](https://www.linkedin.com/in/dimitre-oliveira-7a1a0113a/)
+Date created: 2021/02/25
+Last modified: 2021/02/25
+Description: Converting data to TFRecords files.
+"""
+
+"""
+## Introduction
+
+The TFRecord format is a simple format for storing a sequence of binary records,
+converting your data into TFRecords has many advantages:
+
+- **Efficient storage**: the data may end up taking less space and it can also be partitioned into
+multiple files.
+- **Fast I/O**: the data can be read with parallel I/O operations, useful for TPUs or
+multiple hosts.
+- **Self-contained files**: the data can be read from a single source (e.g.
+by contrast, [COCO](https://cocodataset.org/) stores data into two folders,
+"images" and "annotations").
+
+An important use case of TFRecords is training on TPUs: TPUs are fast enough that
+they usually benefit from optimized I/O operations. In addition, TPUs require
+data to be stored remotely (e.g. on Google Cloud Storage), and storing your data as TFRecords
+makes it easier to the load data without batch-downloading it first.
+
+Performance with TFRecords can be further improved if you use it with the
+[tf.data](https://www.tensorflow.org/guide/data) API.
+
+Here, we will learn how to convert data of different types (image, text, and numeric) into
+TFRecords.
+
+**Reference**
+
+- [TFRecord and tf.train.Example](https://www.tensorflow.org/tutorials/load_data/tfrecord)
+
+
+## Dependencies
+"""
+
+import os
+import json
+import pprint
+import tensorflow as tf
+import matplotlib.pyplot as plt
+
+"""
+## Downloading the data
+"""
+
+root_dir = "datasets"
+tfrecords_dir = "tfrecords"
+images_dir = os.path.join(root_dir, "val2017")
+annotations_dir = os.path.join(root_dir, "annotations")
+annotation_file = os.path.join(annotations_dir, "instances_val2017.json")
+images_url = "http://images.cocodataset.org/zips/val2017.zip"
+annotations_url = (
+    "http://images.cocodataset.org/annotations/annotations_trainval2017.zip"
+)
+
+# Download image files
+if not os.path.exists(images_dir):
+    image_zip = tf.keras.utils.get_file(
+        "images.zip", cache_dir=os.path.abspath("."), origin=images_url, extract=True,
+    )
+    os.remove(image_zip)
+
+# Download caption annotation files
+if not os.path.exists(annotations_dir):
+    annotation_zip = tf.keras.utils.get_file(
+        "captions.zip",
+        cache_dir=os.path.abspath("."),
+        origin=annotations_url,
+        extract=True,
+    )
+    os.remove(annotation_zip)
+
+print("Dataset is downloaded and extracted successfully.")
+
+with open(annotation_file, "r") as f:
+    annotations = json.load(f)["annotations"]
+
+print(f"Number of images: {len(annotations)}")
+
+"""
+## The COCO2017 dataset
+
+We will be using the [COCO2017](https://cocodataset.org/) dataset, because it has many
```

Comment:
Question: should this introduction to why you're using COCO be somewhere before the "downloading the dataset" section?

Reply:
Hi @8bitmp3 , I left this introduction after downloading the data because I wanted to follow this introduction with a display of some annotation samples, so I needed it after the data is downloaded, but alternatively, I could put this introduction before the download section and put this sample as text instead of printing it, what do you think?

Reply:
IMO you can put the first paragraph here under "Download the COCO2017 dataset" and keep the rest here. The title of this section can then be "Contents of the COCO2017 dataset"

Reply:
Sounds good!